lib/bridge/MatrixHandler.js

```diff
@@ -369,7 +369,101 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
+                    'irc.server2';
+        req.log.info(`No valid (old form) admin command, will try new format: ${fmt}`);
+
+        // Assumes commands have the form
+        // irc.server
+        // (COMMAND [arg0 [arg1 [...]]])
+        // (COMMAND [arg0 [arg1 [...]]])
+        // irc.server2
+        // ...
+        let commands = event.content.body;
+
+        try {
+            // Seperate [server+commands, ...]
+
+            let matchBlocks = commands.match(/(\r?\n)?([a-z0-9\.:-]+)(\r?\n([A-Z]+( .*)?))+/g);
+            // Each block looks like this:
+            // irc.server
+            // (COMMAND [arg0 [arg1 [...]]])
+            // (COMMAND [arg0 [arg1 [...]]])
+            // ...
+
+            if (!matchBlocks) {
+                throw new Error(`Badly formed command block(s)`);
+            }
+
+            for (let i = 0; i < matchBlocks.length; i++) {
+                let block = matchBlocks[i];
+
+                let ircDomain = block.match(/^(?:\r?\n)?([a-z0-9\.:-]+)/)[1];
+                block = block.substring(ircDomain.length);
+
+                // set domain of command to ircDomain
+                let server = this.ircBridge.getServer(ircDomain);
+
+                if (!server) {
+                    throw new Error(`IRC domain not found (${ircDomain})`);
+                }
+
+                let commands = block.match(/[A-Z]+(?: ([^\n\r]*))?(?:\r?\n)?/g)
+                // (COMMAND [arg0 [arg1 [...]]])
+                // (COMMAND [arg0 [arg1 [...]]])
+                // ...
+
+                if (!commands) {
+                    throw new Error(`Badly formed command(s)`);
+                }
+
+                for (let j = 0; j < commands.length; j++) {
+                    let commandMatch = commands[j].match(/([A-Z]+)(?: (.*))?/);
+                    // (KEYWORD [arg0 [arg1 [...]]])
+
+                    if (!commandMatch) {
+                        throw new Error(`Malformed command (${commands[i]})`);
+                    }
+
+                    let keyword = commandMatch[1];
+
+                    if (!keyword) {
+                        throw new Error(`Keyword required (${commands[i]})`);
+                    }
+
+                    let argsJoined = commandMatch[2];
+                    let sendArgs = [];
+
+                    // Arguments are optional
+                    if (!argsJoined) {
+                        argsJoined = '';
+                    }
+                    else {
+                        sendArgs = argsJoined.split(' ');
+                    }
+
+                    sendArgs.unshift(keyword);
+
+                    let bridgedClient = yield this.ircBridge
+                        .getBridgedClient(server, event.user_id);
+
+                    if (!bridgedClient.unsafeClient) {
+                        throw new Error('Possibly disconnected');
+                    }
+
+                    Client.prototype.send.apply(bridgedClient.unsafeClient, sendArgs);
```

Comment:
uhhh what? Why not just `bridgedClient.unsafeClient.send.apply(bridgedClient.unsafeClient, sendArgs)`?